src/cli.rs

```diff
@@ -159,6 +159,27 @@ pub struct RootOpts {
     )]
     pub internal_log_rate_limit: u64,
 
+    /// Set the duration in seconds to wait for graceful shutdown after SIGINT or SIGTERM are received.
+    /// After the duration has passed, Vector will force shutdown.
+    #[arg(
+        long,
+        default_value = "60",
+        env = "VECTOR_GRACEFUL_SHUTDOWN_DURATION",
+        group = "graceful-shutdown-duration"
+    )]
+    pub graceful_shutdown_duration: NonZeroU64,
```

Comment:
Typically we've suffixed time options with the unit they expect. In this case it'd be `graceful_shutdown_duration_secs`. Given that feels a bit repetitive, I'd suggest `graceful_shutdown_secs`. We have some documentation for this for [configuration fields](https://github.com/vectordotdev/vector/blob/master/docs/specs/configuration.md#option-naming). We don't have similar documentation for flags, but I think applying the units suffix rules makes sense here.

Reply:
implemented a new name here, let me know if it looks good (we also want alignment internally): https://github.com/vectordotdev/vector/pull/17479/commits/15e28387351aa07587420ec5a01bf9bbb83a7d21